### PR TITLE
updated package setup

### PR DESCRIPTION
**Pull request by Goldziher**
Rebased version of the previous PR:

add pre-commit
improve the travis setup
cleanup and update the md files
improve the project folder structure
add stubgen scripts
update setup.py

### CODE REVIEW

**Comment by Goldziher** [comment]
Seems that there are some problems that will require more work. Next weekend it is :).

**Comment by Goldziher** [comment]
this PR is ready for review @sobolevn

**Comment by kszmigiel** [comment]
@Goldziher thank you for your work!

**Comment by Goldziher** [comment]
@sobolevn i rebased.

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-pytest-mypy-plugins==1.6.1
+wheel
+mypy==0.790
+requests==2.24.0
+coreapi==2.3.3
+typing-extensions==3.7.4.3
+gitpython==3.1.9
+pre-commit==2.7.1
+pytest==6.1.1
+pytest-mypy-plugins==1.6.0
```
Comment (sobolevn) [inline dev-requirements.txt]: It should be `1.6.1`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot!

File: dev-requirements.txt (hunk 1 above)
Comment (sobolevn) [reply dev-requirements.txt]: Then the test failure will go away.

Comment (Goldziher) [reply dev-requirements.txt]: done

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
As always, @Goldziher, great work! 👍

**Pull request merged**